packages/smooth_app/lib/data_models/onboarding_data_product.dart

```diff
@@ -42,7 +42,7 @@ class OnboardingDataProduct extends AbstractOnboardingData<Product> {
           language: ProductQuery.getLanguage(),
           country: ProductQuery.getCountry(),
         ),
-      );
+      ).timeout(const Duration(seconds: 20));
```

Comment:
20 second is long.
I would suggest something shorter like 5 seconds, and a "wink" message like "Oops... I'm afraid we'll have to display data in English, sorry about that ;)"